torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -5778,7 +5778,14 @@ def test_broadcast_object_list(self):
                 else [None for _ in COLLECTIVES_OBJECT_TEST_LIST]
             )
 
-            # Single object test
+            # Single object test with dist_device_specified
+            single_obj_list = [objects[0]]
+            if self.rank != src_rank:
+                self.assertNotEqual(single_obj_list[0], COLLECTIVES_OBJECT_TEST_LIST[0])
+            dist.broadcast_object_list(single_obj_list, src=0, group=None, dist_device=torch.device("cpu"))
+            self.assertEqual(single_obj_list[0], COLLECTIVES_OBJECT_TEST_LIST[0])
```

Comment:
can we also have a test where we specify a non-current CUDA device for both nccl and gloo backends?

Reply:
updated current test to use next rank